### DISCUSSION TITLE
Lime3DS, citra is a dead end

### DISCUSSION THREAD

**Discussion by Exarkuniv** [Exarkuniv/RetroPie-Extra#220] [Ideas]
https://github.com/Lime3DS/Lime3DS?tab=readme-ov-file

this does build, 

citra does not work, even with the new links. 
it does not like the CPU, so not even going to try anymore

**Comment by RoseEverdeen** [comment]
[citra.sh.zip](https://github.com/user-attachments/files/16069554/citra.sh.zip)
I located this script for install citra from https://github.com/NGC602/citra_yuzu.  I've tweaked it a bit to point to the new repo and ran it on ubuntu cinnamon 22.04 and i managed to compile citra.
I also noticed on the new repo they used git clone --recursive https://github.com/PabloMK7/citra
cd citra

git submodule sync
git submodule foreach --recursive git reset --hard
git submodule update --init --recursive

I'm not very versed with Linux but thought I'd let you know in case it helps.

**Comment by Exarkuniv** [comment]
i did try that one also, and it would not work for me. which is why i looked into lime, since that one compiled for me very easyly

but i will try your script

**Comment by Exarkuniv** [comment]
so i tested the schipt, and it did install, 

i just need to get some roms and test it

loads the GUI, but GPU drivers are not supported

**Comment by StonedEdge** [comment]
This compiled for me too on an RK3588 running Armbian (basically Ubuntu). I wonder if we can get the libretro core working?

**Comment by Exarkuniv** [comment]
I have tried that. And I got nowhere. Other the. Finding more ways it will not work. 

Which is why I'm going to try lime.

But if you want to give a shot. Go for it. I'll gladly update the script if you clget it working.

**Comment by StonedEdge** [comment]
No problem. I'm running an RK3588 and I was able to get the standalone emulator working by forcing the GL version to be 4.3, but since my GPU only supports Open GL 3.3 at the moment, was hoping to be able to use the libretro core.

**Comment by Exarkuniv** [comment]
Yes that's the best way to go. 
For the pi5 I don't think there's a way to force the openGL without the chance to breaking something else

**Comment by Exarkuniv** [comment]
found this. might be worth looking into 
https://github.com/jonian/libretro-panda3ds

**Comment by DTEAM-1** [comment]
I'm not good to create .sh file. I see this : https://github.com/PabloMK7/citra   for batocera  . They use 
      https://github.com/PabloMK7/citra

**Comment by Exarkuniv** [comment]
Yes i know of that, and i have tried to use it, and again didn't work. might try again later but for now im going to use Lime and go from there

**Comment by DTEAM-1** [comment]
They fixed something with OpenGL on july 10 under under aarch64 . Don't know if it's helpful. 
https://github.com/Lime3DS/Lime3DS/commit/b240dac871292c9bf04e581a9bcf7021861858e7
https://github.com/Lime3DS/Lime3DS/issues/124

I tried to write a Shell script for Lime3DS with no success. I don't have enough knowledge for now. I also tried a [flatpack ](https://github.com/flathub/org.citra_emu.citra/tree/master)with no success.

**Comment by Exarkuniv** [comment]
they fixed the links in citra. this might be working again

**Comment by Exarkuniv** [comment]
trying script from here
https://retropie.org.uk/forum/topic/36807/suggestions-for-ports-4/6?_=1737042066443

fails at 96%

```
[ 96%] Building CXX object src/lime_qt/CMakeFiles/lime_qt.dir/bootmanager.cpp.o
/home/pi/RetroPie-Setup/tmp/build/lime3ds/src/lime_qt/bootmanager.cpp:42:10: fatal error: qpa/qplatformnativeinterface.h: No such file or directory
   42 | #include <qpa/qplatformnativeinterface.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[ 96%] Building CXX object src/lime_qt/CMakeFiles/lime_qt.dir/camera/camera_util.cpp.o
gmake[2]: *** [src/lime_qt/CMakeFiles/lime_qt.dir/build.make:610: src/lime_qt/CMakeFiles/lime_qt.dir/bootmanager.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:2669: src/lime_qt/CMakeFiles/lime_qt.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
In function ‘lexicographical_compare_three_way’,
    inlined from ‘operator<=>’ at /usr/include/c++/12/bits/stl_vector.h:2056:52,
    inlined from ‘operator()’ at /usr/include/c++/12/bits/stl_function.h:408:20,
    inlined from ‘_M_lower_bound’ at /usr/include/c++/12/bits/stl_tree.h:1951:29,
    inlined from ‘find’ at /usr/include/c++/12/bits/stl_tree.h:2531:36:
/usr/include/c++/12/bits/stl_algobase.h:1843:41: warning: ‘__builtin_memcmp’ specified bound [9223372036854775808, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Wstringop-overread]
 1843 |                       = __builtin_memcmp(&*__first1, &*__first2, __len) <=> 0;
      |                                         ^
[ 96%] Built target tests
gmake: *** [Makefile:146: all] Error 2
/home/pi/RetroPie-Setup
```

**Reply by DTEAM-1** [reply]
On my side, it works. Don't know why.

**Reply by Exarkuniv** [reply]
what build are you using. is this a basic one, or one with vulcan?

**Reply by DTEAM-1** [reply]
I use xinit to load it, but inside the software, vulkan is available but it don't work. It suppose to be the best option . Only the slow software option works

**Reply by DTEAM-1** [reply]
the one with vulkan. It's suppose to be that one,

**Reply by Exarkuniv** [reply]
ok, well then it will be a bit before i mess more with this. i only have one SD and i have the basic load on it. since im retesting everything. 

im going with that is why im getting errors.

**Reply by Exarkuniv** [reply]
so i added a few missing depends so now the script do work and works well. 
as you said it is very slow and needs vulcan. but thats nothing on the script. so thanks for getting it this far

**Reply by DTEAM-1** [reply]
Could you share you share your script in the chat please. I'll modify mine.

**Reply by Exarkuniv** [reply]
```
function depends_lime3ds() {
    if compareVersions $__gcc_version lt 7; then
        md_ret_errors+=("Sorry, you need an OS with gcc 7.0 or newer to compile lime3ds")
        return 1
    fi

    # Additional libraries required for running
    local depends=(build-essential cmake clang clang-format libc++-dev libsdl2-dev qt6-l10n-tools qt6-tools-dev qt6-tools-dev-tools  qt6-base-dev qt6-base-private-dev qt6-multimedia-dev libfdk-aac-dev ffmpeg libgl-dev  libswscale-dev libavformat-dev libavcodec-dev libavdevice-dev )
    getDepends "${depends[@]}"
}
```
also i needed to add the extensions of the roms so it would show on my ES
```
function configure_lime3ds() {

    mkRomDir "3ds"
	ensureSystemretroconfig "3ds"
	    local launch_prefix
    isPlatform "kms" && launch_prefix="XINIT-WM:"

    addEmulator 0 "$md_id" "3ds" "$launch_prefix$md_inst/lime3ds"
  	#addEmulator 2 "$md_id-qt" "3ds" "$md_inst/lime3ds-room %ROM%"
	
    addSystem "3ds" "3ds" ".3ds .zip"

}
```
